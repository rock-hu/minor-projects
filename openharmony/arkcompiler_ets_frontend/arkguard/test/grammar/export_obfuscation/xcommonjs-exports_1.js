/*
 * Copyright (c) 2023 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

let exportVar = 1;

exports.myExportVar = exportVar;
exports.myVariable = 2;
exports.myFunction = function() {
  return 'Hello, world!';
};

class CommonjsClass1 {
  id3 = '00001';
  info3 = {name3: 'jack3', age3: 12};
}

exports.CommonjsClass1 = CommonjsClass1;

exports.noNameClass = class {
  stuID = 11
  stuAge = 15
  getAge(){
    return this.stuAge;
  }
}