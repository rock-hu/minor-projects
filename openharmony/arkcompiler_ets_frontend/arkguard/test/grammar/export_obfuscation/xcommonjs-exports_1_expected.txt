/*
 * Copyright (c) 2023 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
let e5 = 1;
exports.o3 = e5;
exports.q3 = 2;
exports.r3 = function () {
    return 'Hello, world!';
};
class f5 {
    s3 = '00001';
    d4 = { e4: 'jack3', f4: 12 };
}
exports.f5 = f5;
exports.h4 = class {
    i4 = 11;
    j4 = 15;
    r4() {
        return this.j4;
    }
};
